profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,568 @@
+        //add a NIC to the server and assign an IP to the server
+
+        int lanId = DEFAULT_LAN_ID;
+        if (options.getNetworks() != null) {
+            try {
+                String networkId = Iterables.get(options.getNetworks(), 0);
+                lanId = Integer.valueOf(networkId);
+            } catch (Exception ex) {
+                logger.warn("no valid network id found from options. using default id='%d'", DEFAULT_LAN_ID);
+            }
+        }
+        IpBlock ipBlock = api.ipBlockApi().create(IpBlock.Request.creatingBuilder()
+                .properties(PropertiesRequest.create(name + " block", template.getLocation().getParent().getId(), 1))
+                .build());
+
+        Nic nic = api.nicApi().create(Nic.Request.creatingBuilder()
+                .dataCenterId(dataCenterId)
+                .name("jclouds" + name)
+                .dhcp(Boolean.TRUE)
+                .lan(lanId)
+                .ips(ipBlock.properties().ips())
+                .firewallActive(Boolean.FALSE)
+                .serverId(serverId).
+                build());
+
+        waitNICUntilAvailable.apply(NicRef.create(dataCenterId, serverId, nic.id()));
+
+//        connect the rest of volumes to server;delete if fails
+        final int volumeCount = volumeIds.size();
+        for (int j = 1; j < volumeCount; j++) { // skip first; already connected
+            final String volumeId = volumeIds.get(j);
+            try {
+                logger.trace("<< connecting volume '%s' to server '%s'", volumeId, serverId);
+                provisioningManager.provision(jobFactory.create(group, new Supplier<Object>() {
+
+                    @Override
+                    public Object get() {
+                        return api.serverApi().attachVolume(
+                                Server.Request.attachVolumeBuilder()
+                                .dataCenterId(dataCenterId)
+                                .serverId(serverId)
+                                .volumeId(volumeId)
+                                .build()
+                        );
+                    }
+                }));
+
+                logger.trace(">> volume connected.");
+            } catch (Exception ex) {
+                // delete unconnected volume
+                logger.warn(ex, ">> failed to connect volume '%s'. deleting..", volumeId);
+                destroyVolume(volumeId, dataCenterId);
+            }
+        }
+        // Last paranoid check
+        waitDcUntilAvailable.apply(dataCenterId);
+
+        LoginCredentials serverCredentials = LoginCredentials.builder()
+                .user(loginUser)
+                .password(password)
+                .build();
+        String serverInDataCenterId = DataCenterAndId.fromDataCenterAndId(dataCenterId, serverId).slashEncode();
+        rebootNode(serverInDataCenterId);
+        waitServerUntilAvailable.apply(ServerRef.create(dataCenterId, serverId));
+
+        try {
+            Thread.sleep(90000);
```

Comment:
Also, why the need for 90-second sleep here? If the provisioning process still not done here (*sigh*), adjust instead the default values of the `waitServerUntilAvailable` predicate or datacenter, whichever applicable.